chicago/events.py

```diff
@@ -29,40 +30,86 @@ def eventPages(self, event_url, search_type='all') :
             return self.pages(page)
 
 
-    def scrape(self):
+    def scrape(self, follow_links=True):
         for page in self.eventPages(EVENTSPAGE):
             events_table = page.xpath("//table[@class='rgMasterTable']")[0]
             for events, headers, rows in self.parseDataTable(events_table) :
-                print(events)
+                if follow_links and type(events['Meeting\xa0Details']) == dict :
+                    detail_url = events['Meeting\xa0Details']['url']
+                    meeting_details = self.lxmlize(detail_url)
+
+                    agenda_table = meeting_details.xpath(
+                        "//table[@id='ctl00_ContentPlaceHolder1_gridMain_ctl00']")[0]
+                    agenda = self.parseDataTable(agenda_table)
+
+                    
                 location_string = events[u'Meeting\xa0Location']
                 location_list = location_string.split('--')
                 location = ', '.join(location_list[0:2])
 
-                status_string = location_list[-1].split('Chicago, Illinois')
-                if len(status_string) > 1 and status_string[1] :
-                    status = status_string[1].lower()
-                    if status not in ['cancelled', 'tentative', 'confirmed', 'passed'] :
-                        print(status)
-                        status = 'confirmed'
-                else :
-                    status = 'confirmed'
-
-
-
                 when = events[u'Meeting\xa0Date']
                 time_string = events[u'Meeting\xa0Time']
                 event_time = datetime.datetime.strptime(time_string,
                                                         "%I:%M %p")
                 when = when.replace(hour=event_time.hour)
+                when = when.replace(tzinfo=pytz.timezone("US/Central"))
+
+                
+
+                status_string = location_list[-1].split('Chicago, Illinois')
+                if len(status_string) > 1 and status_string[1] :
+                    status_text = status_string[1].lower()
+                    if any(phrase in status_text 
+                           for phrase in ('rescheduled to',
+                                          'postponed to',
+                                          'reconvened to',
+                                          'recessed',
+                                          'cancelled',
+                                          'new date and time',
+                                          'rescheduled indefinitely',
+                                          'rescheduled for')) :
+                        status = 'cancelled'
+                    elif status_text in ('rescheduled') :
+                        status = 'cancelled'
+                    else :
+                        print(status_text)
+                elif datetime.datetime.utcnow().replace(tzinfo = pytz.utc) > when :
+                    status = 'confirmed'
+                else :
+                    status = 'passed'
```

Comment:
Yeah, for passed, totally. Let's get a bug open on that, and leave it at `confirmed`